File: backend/src/main/java/com/hotelmanagement/service/EmployeeService.java
package com.hotelmanagement.service;

import java.util.List;

import com.hotelmanagement.dto.request.EmployeeRequest;
import com.hotelmanagement.dto.response.EmployeeResponse;
import com.hotelmanagement.enums.EmployeePosition;

public interface EmployeeService {

    // Create
    EmployeeResponse createEmployee(EmployeeRequest request);

    // Update
    EmployeeResponse updateEmployee(Integer employeeId, EmployeeRequest request);

    // Delete
    void deleteEmployee(Integer employeeId);

    // Get by ID
    EmployeeResponse getEmployeeById(Integer employeeId);

    // Get all
    List<EmployeeResponse> getAllEmployees();

    // Search by last name
    List<EmployeeResponse> findByLastName(String lastName);

    // Search by position
    List<EmployeeResponse> findByPosition(EmployeePosition position);

}

File: frontend/src/services/employeeService.js
import api from "../api/api";

// =====================================================
// GET ALL EMPLOYEES
// =====================================================

export const getEmployees = () => {

    return api.get("/employees");

};


// =====================================================
// GET EMPLOYEE BY ID
// =====================================================

export const getEmployeeById = (id) => {

    return api.get(`/employees/${id}`);

};


// =====================================================
// CREATE EMPLOYEE
// =====================================================

export const createEmployee = (employee) => {

    return api.post("/employees", employee);

};


// =====================================================
// UPDATE EMPLOYEE
// =====================================================

export const updateEmployee = (id, employee) => {

    return api.put(`/employees/${id}`, employee);

};


// =====================================================
// DELETE EMPLOYEE
// =====================================================

export const deleteEmployee = (id) => {

    return api.delete(`/employees/${id}`);

};


// =====================================================
// FIND EMPLOYEES BY LAST NAME
// =====================================================

export const getEmployeesByLastName = (lastName) => {

    return api.get(
        `/employees/lastname/${lastName}`
    );

};


// =====================================================
// FIND EMPLOYEES BY POSITION
// =====================================================

export const getEmployeesByPosition = (position) => {

    return api.get(
        `/employees/position/${position}`
    );

};